src/deserializeSchema.ts

```diff
@@ -0,0 +1,486 @@
+        case ParameterType.I8:
+        case ParameterType.I16:
+        case ParameterType.I32:
+        case ParameterType.I64:
+        case ParameterType.I128:
+        case ParameterType.Amount:
+        case ParameterType.AccountAddress:
+        case ParameterType.ContractAddress:
+        case ParameterType.Timestamp:
+        case ParameterType.Duration:
+            return { typeTag: tag };
+        case ParameterType.Pair:
+            return {
+                typeTag: tag,
+                ofLeft: deserialType(source),
+                ofRight: deserialType(source),
+            };
+        case ParameterType.List:
+        case ParameterType.Set:
+            return {
+                typeTag: tag,
+                sizeLength: deserialUint8(source),
+                of: deserialType(source),
+            };
+        case ParameterType.Map:
+            return {
+                typeTag: tag,
+                sizeLength: deserialUint8(source),
+                ofKeys: deserialType(source),
+                ofValues: deserialType(source),
+            };
+        case ParameterType.Array:
+            return {
+                typeTag: tag,
+                size: deserialUint32(source),
+                of: deserialType(source),
+            };
+        case ParameterType.Struct:
+            return {
+                typeTag: tag,
+                fields: deserialFields(source),
+            };
+        case ParameterType.Enum:
+            return {
+                typeTag: tag,
+                variants: deserialArrayFn<[string, Fields]>(
+                    deserialTupleFn<string, Fields>(
+                        deserialString,
+                        deserialFields
+                    )
+                )(source),
+            };
+        case ParameterType.String:
+        case ParameterType.ContractName:
+        case ParameterType.ReceiveName:
+            return {
+                typeTag: tag,
+                sizeLength: deserialUint8(source),
+            };
+        default:
+            throw new Error(`unsupported type tag: ${tag}`);
+    }
+}
+
+/**
+ * {@link Fields} tag.
+ */
+export enum FieldsTag {
+    /**
+     * Represents named fields such as in
+     * `struct RGB { r: u8, g: u8, b: u8 }`.
+     */
+    Named = 0,
+    /**
+     * Represents unnamed (anonymous) struct fields such as in
+     * `struct Point { u32, u32 }`.
+     */
+    Unnamed,
+    /**
+     * Represents lack of fields in a struct or an enum, as is the case
+     * with `Cat` in `enum Animal { Cat, Dog, Human }`.
+     */
+    None,
+}
+
+/**
+ * Rust flavored struct/enum fields.
+ */
+export type Fields =
+    | NamedFields
+    | UnNamedFields
+    | {
+          fieldsTag: FieldsTag.None;
+      };
+/**
+ * Name fields if the type is Struct/enum
+ */
+export type NamedFields = {
+    fieldsTag: FieldsTag.Named;
+    contents: [string, Type][];
+};
+
+/**
+ * UnName fields if the type is Struct/enum
+ */
+export type UnNamedFields = {
```

Comment:
```suggestion
 * Unnamed fields if the type is Struct/enum
 */
export type UnnamedFields = {
```
Unnamed is actually a word :thinking: